### ISSUE TITLE
DeepSSM in Studio - Training

### ISSUE THREAD

**Issue by jadie1** [SCIInstitute/ShapeWorks#1373]
Need a tab in Studio for training DeepSSM

Parameters:

- Epochs
- Learning rate
- Decay learning rate - true or false 
- Fine tuning - true or false
- Epochs

Run training button - progress bar at bottom

Display (updated every 10 epochs): 

- Model training dashboard with train and validation loss
- Examples of reconstructed meshes with error heat map - best, median, and worst in validation

**Comment by akenmorris** [comment]
@jadie1 , where can I get access to the training dashboard?  Is it written out after every epoch?  How do I reconstruct meshes with heat maps?

**Comment by jadie1** [comment]
Currently it just prints out the info to the command line. The default is every epoch but he `trainer` `val_freq` parameter can be set to have it test on the validation set and print less often than every epoch (for example if set to 2 it prints every other) . For now in studio I think we could just set the frequency it updates to something constant like 10.  

To get the distance meshes in Python you call:
```
import shapeworks as sw
mesh1 = sw.Mesh(path_to_mesh1)
mesh2 = sw.Mesh(path_to_mesh2)
mesh1.distance(mesh2).write(path_to_out_distance_mesh1)
```

**Comment by akenmorris** [comment]
Ok, this is the dashboard you want to see somewhere?

```
Epoch	  LR	  Train_Err	  Train_Rel_Err	  Val_Err	  Val_Rel_Err	  Sec
1.00000	  0.00100	  2.07978	  5.10448	  8.79684	  1040.82276	  11.55580
2.00000	  0.00098	  68.11378	  5475.04883	  12.66105	  2341.53674	  11.90325
3.00000	  0.00097	  6.27155	  46.41597	  84.35498	  122743.24353	  11.96991
```

Should we have this drawn as a chart?  Or just the table?  Or Both?

For the meshes, where are they written?  I haven't found any meshes being written as part of the training.

**Comment by jadie1** [comment]
I think drawing them as a chart would be more user friendly. We could just plot the training and validation relative error over epochs. In the use case the meshes with error heat maps are not written until after training. During evaluation they are generated for the test set. We probably want to adapt this so that the best worst and average examples from the validation set are displayed.  I can write a Python function for this if that'd be helpful.

**Comment by akenmorris** [comment]
Yeah, a python function that does this would be great.  We should draw the chart in python with matplotlib as well.

I will add an epoch hook to send control back to Studio and these things can be updated.
